docs/_renderer.py

```diff
@@ -282,3 +235,48 @@ def read_file(file: str | Path, root_dir: str | Path | None = None) -> FileConte
         "content": file_content,
         "type": type,
     }
+
+
+def check_if_missing_expected_example(el, converted):
+    if re.search(r"(^|\n)(#{2,6} Examples\n|Examples\n------)", converted):
+        # Manually added examples are fine
+        return
+
+    if not el.canonical_path.startswith("shiny"):
+        # Only check Shiny objects for examples
+        return
+
+    if hasattr(el, "decorators") and "no_example" in [
+        d.value.canonical_name for d in el.decorators
+    ]:
+        # When an example is intentionally omitted, we mark the fn with `@no_example`
+        return
+
+    if not el.is_function:
+        # Don't throw for things that can't be decorated
+        return
+
+    if not el.is_explicitely_exported:
+        # Don't require examples on "implicitly exported" functions
+        # In practice, this covers methods of exported classes (class still needs ex)
+        return
+
+    # TODO: Remove shiny.express from no_req_examples when we have examples ready
+    no_req_examples = ["shiny.express", "shiny.experimental"]
+    if any([el.target_path.startswith(mod) for mod in no_req_examples]):
+        return
+
+    raise RuntimeError(
+        f"{el.name} needs an example, use `@add_example()` or manually add `Examples` section:\n"
+        + (f"> file     : {el.filepath}\n" if hasattr(el, "filepath") else "")
+        + (f"> target   : {el.target_path}\n" if hasattr(el, "target_path") else "")
+        + (f"> canonical: {el.canonical_path}" if hasattr(el, "canonical_path") else "")
+    )
+
+
+def remove_sphinx_comments(converted: str) -> str:
+    """
+    Sphinx allows `..`-prefixed comments in docstrings, which are not valid markdown.
+    This function removes them.
```

Comment:
```suggestion
    Sphinx allows `..`-prefixed comments in docstrings, which are not valid markdown.
    This can be useful when we want to include comments that shouldn't appear in the
    hover card or in the rendered docs. This function removes sphinx comments, but it
    does not cover sphinx directives (don't use those!).
```